### PR TITLE
docs - add info about per-greenplum-user pxf configuration

### PR DESCRIPTION
**Pull request by lisakowen**
can now configure greenplum users in PXF server definitions.  in this PR:
- update "About PXF Server Configuration" page:
    - include per-user info
    - reorg sections
    - new sections on user config, config property precedence
- misc edits to existing "override" references

doc review site link to "About PXF Server Configuration" page: http://docs-lisa-pxf-usercfg.cfapps.io/6-0/pxf/cfg_server.html

### CODE REVIEW

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb [hunk 1]
```diff
@@ -97,20 +155,27 @@ CREATE EXTERNAL TABLE pxf_ext_hdfs(location text, miles int)
...
+1. A property that you configure in `<server_name>/<greenplum_user_name>-user.xml` overrides any setting of the property elsewhere.
+2. A property that is specified via custom options in the `CREATE EXTERNAL TABLE` command `LOCATION` clause overrides any setting of the property in a PXF server configuration.
+3. Properties that you configure in the `<server_name>` PXF server definition identify the default property values.
+
+These precedence rules allow you create a single external table that can be accessed by multiple Greenplum Database users, each with their own unique remote data store user credentials.
 
-## <a id="override"></a>Overriding the Server Configuration
 
-Some PXF connectors (S3, JDBC) allow you to override a server configuration by directly specifying certain properties via custom options in the `CREATE EXTERNAL TABLE` command `LOCATION` clause. Refer to [Overriding the S3 Server Configuration](access_objstore.html#s3_override) and [Overriding the JDBC Server Configuration](jdbc_cfg.html#override) for additional information.
+## <a id="srv-cfg-update"></a>Adding or Updating Server and User Configurations
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: Merge this with the "Adding" procedure above?

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb [hunk 2]
```diff
@@ -4,44 +4,27 @@ title: About PXF Server Configuration
 
 This topic provides an overview of PXF server configuration. To configure a server, refer to the topic specific to the connector that you want to configure.
 
-You read from or write data to an external data store via a PXF connector. To access an external data store, you must provide the server location. You may also be required to provide client access credentials and other external data store-specific properties. PXF simplifies configuring access to external data stores by:
+You read from or write data to a remote data store via a PXF connector. To access a remote data store, you must provide the server location. You may also be required to provide client access credentials and other remote data store-specific properties. PXF simplifies configuring access to remote data stores by:
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: to distinguish from "external table"; there were alot of external's on the page.  i can certainly change the "remote"s back to "external"s.

**Comment by lisakowen** [comment]
@dyozie - changed titles, reorg'd a bit, and pushed a new review site.  what do you think?

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb [hunk 3]
```diff
@@ -72,14 +55,104 @@ For example, the contents of the `s3-site.xml` template file follow:
...
+## <a id="default"></a>About the Default Server
+
+PXF defines a special server named `default`. When you initialize PXF, it automatically creates a `$PXF_CONF/servers/default/` directory. This directory, initially empty, identifies the default PXF server configuration. You can configure and assign the default PXF server to any remote data source. For example, you can assign the PXF default server to a Hadoop cluster, or to a MySQL database that your users frequently access.
+
+PXF automatically uses the `default` server configuration if you omit the `SERVER=<server_name>` setting in the `CREATE EXTERNAL TABLE` command `LOCATION` clause.
+
+**Note**: You *must* configure a Hadoop server as the PXF `default` server when your Hadoop cluster utilizes Kerberos authentication.
+
+
+## <a id="cfgproc"></a>About the Server Configuration Procedure
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: This topic seems closes to a process topic, so maybe title it "Configuring a Server" or similar?  You could rename the file itself "Configuring PXF Servers" to differentiate if necessary. 

This section should probably also end by providing direct links to actual server configuration procedures.

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb [hunk 4]
```diff
@@ -72,14 +55,104 @@ For example, the contents of the `s3-site.xml` template file follow:
...
+3. Copy template or other configuration files to the new server directory.
+4. Fill in appropriate default values for the properties in the template file.
+5. Add any additional configuration properties and values required for your environment.
+6. Configure one or more users for the server configuration as described in [About Configuring a PXF User](#usercfg).
+7. Synchronize the server and user configuration to the Greenplum Database cluster.
+8. Publish the PXF server names to your Greenplum Database end users as appropriate.
+
+**Note**: You must re-sync the PXF configuration to the Greenplum Database cluster after you add or update PXF server configuration.
+
+## <a id="usercfg"></a>About Configuring a PXF User
```
Comment (dyozie) [inline gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: Same reasoning here - suggest the title "Configuring a PXF User"

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: thanks for reviewing, @frankgh !

yes, we should be consistent with the terminology.  right now, what distinguishes the use of an external table vs a foreign table is whether or not the component is implemented as a foreign data wrapper.  we should revisit this terminology question when pxf provides FDWs to access external/foreign data.  i am afraid that if we refer to the data as foreign now, there may be some expectation of PXF being an FDW now.

i changed the references from remote back to external.

File: gpdb-doc/markdown/pxf/cfg_server.html.md.erb [hunk 5]
```diff
@@ -1,47 +1,28 @@
...
+A PXF *Server* definition is simply a named configuration that provides access to a specific remote data store. A PXF server name is the name of a directory residing in `$PXF_CONF/servers/`. The information that you provide in a server configuration is connector-specific. For example, a PXF JDBC Connector server definition may include settings for the JDBC driver class name, URL, username, and password. You can also configure connection- and session-specific properties in a JDBC server definition.
 
-You configure a server definition for each external data store that you will permit Greenplum Database users to access. For example, if you require access to two Hadoop clusters, you will create a PXF Hadoop server configuration for each cluster. If you require access to an Oracle and a MySQL database, you will create a PXF JDBC server configuration for each database/user combination.
+PXF provides a server template file for each connector; this template identifies the minimum set of properties that you must configure to use the connector.
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: yes.  and some properties may not be set if using <user>-user.xml files.  i am sure there are other cases.  i will address this (template file not always required) when i investigate and add the docs for s3 support of IAM roles.

changed minimum to "typical".

[the remark this answers is not in the document]
Comment (frankgh) [reply gpdb-doc/markdown/pxf/cfg_server.html.md.erb]: thank you!

**Pull request merged**